app/services/idv/session.rb

```diff
@@ -291,6 +290,11 @@ def desktop_selfie_test_mode_enabled?
       IdentityConfig.store.doc_auth_selfie_desktop_test_mode
     end
 
+    def idv_consent_given?
+      !!(session[:idv_consent_given_at] &&
+        session[:idv_consent_given_at] > (Time.zone.now - 1.month))
```

Comment:
On the one hand, I don't think it should be realistically _possible_ for a user to end up with a consent_given_at timestamp that's a month old, since they would have keep their session active for a month. I think I would vote to limit this to just the presence of the attribute and not check its value. "Is this session suspiciously old" feels like a different thing to try and detect / mitigate.

```suggestion
      !!(session[:idv_consent_given_at]
```